src/string.cr

```diff
@@ -3829,58 +3829,90 @@ class String
     Array.new(bytesize) { |i| to_unsafe[i] }
   end
 
-  def inspect(io)
+  # Pretty prints `self` into the given printer.
+  def pretty_print(pp : PrettyPrint) : Nil
+    pp.text(inspect)
+  end
+
+  # Returns a representation of `self` using character escapes for special characters and wrapped in quotes.
+  # ```crystal
```

Comment:
Please keep blank line before code example for readability.
Also, `crystal` here is not needed, code markers are enough.